src/pds_doi_service/core/actions/test/reserve_test.py

```diff
@@ -95,14 +96,15 @@ def test_reserve_xlsx_and_submit(self):
             'force': True
         }
 
-        o_doi_labels = self._action.run(**reserve_args)
+        o_doi_label = self._action.run(**reserve_args)
+
+        dois, errors = DOIOstiJsonWebParser.parse_dois_from_label(o_doi_label)
 
-        for o_doi_label in o_doi_labels:
-            dois, errors = DOIOstiJsonWebParser.parse_dois_from_label(o_doi_label)
+        self.assertEqual(len(dois), 3)
+        self.assertEqual(len(errors), 0)
 
-            self.assertEqual(len(dois), 1)
-            self.assertEqual(len(errors), 0)
-            self.assertTrue(dois[0].status == DoiStatus.Reserved)
+        for doi in dois:
+            self.assertTrue(doi.status == DoiStatus.Reserved)
```

Comment:
7th; the superclass could leave it to subclasses to tell the expected status, number, error length, etc., while it does the "heavy lifting" of parsing the DOIs and making the assertions.